### DISCUSSION TITLE
Inserção de Master-Detail

### DISCUSSION THREAD

**Discussion by traozin** [viniciussanchez/dataset-serialize#151] [Q&A] [answered]
Existe alguma opção de pegar o id do Master e armazenar no Detail durante uma inserção? Estamos tentando inserir, mas, ao fazê-lo, a referência ao id do Master no Detail está entrando como NULL.

**Comment by viniciussanchez** [comment, marked as answer]
Sim, só criar um mestre detalhe nos dataset. Automaticamente deveria funcionar

**Reply by traozin** [reply]
Ok, muito obrigado!

**Comment by ds-sampaio** [comment]
Aproveitando a dúvida do colega, é possível fazer com o zeos no lazarus? eu ate fiz um exemplo e ao gravar recebo uma mensagem de erro onde me informa que a referencia não esta sendo gravada na tabela detail, sabe me informar se é necessário fazer alguma configuração?
Estou usando o Zeos do Lazarus

**Reply by viniciussanchez** [reply]
Até o momento não fiz mestre detalhe ainda no Lazarus para testar
